subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestExecutionScheduler.kt

```diff
@@ -23,50 +24,49 @@ internal class TestExecutionScheduler(
 
     private val states: MutableMap<TestCase, TestExecutionState> = mutableMapOf()
 
-    fun start(
-        requests: List<TestRunRequest>,
-        scope: CoroutineScope
-    ) {
-        scope.launch(dispatcher + CoroutineName("test-state-verdict")) {
-            for (result in intentionResults) {
-                val state = requireNotNull(states[result.intention.action.test]) {
-                    "Can't find state for $result"
-                }
-                when (val verdict = state.verdict(result.actionResult.testCaseRun)) {
-                    is TestExecutionState.Verdict.SendResult ->
-                        results.send(
-                            TestRunResult(
-                                request = state.request,
-                                result = verdict.results
+    suspend fun start(requests: List<TestRunRequest>) {
+        with(CoroutineScope(coroutineContext + dispatcher)) {
```

Comment:
Every `suspend fun` has an implicit `coroutineContext`. So now I use that context for  creating scope and `launch` coroutines there